src/Http/Owin/src/DictionaryStringArrayWrapper.cs

```diff
@@ -74,4 +76,40 @@ bool IDictionary<string, string[]>.TryGetValue(string key, out string[] value)
         value = default(StringValues);
         return false;
     }
+
+    public struct Enumerator : IEnumerator<KeyValuePair<string, string[]>>, IEnumerator
+    {
+        private IEnumerator<KeyValuePair<string, StringValues>> _inner;
+        private KeyValuePair<string, string[]> _current;
+
+        internal Enumerator(IDictionary<string, StringValues> inner)
+        {
+            _inner = inner.GetEnumerator();
+            _current = default;
+        }
+
+        public void Dispose()
+        {
+            _inner?.Dispose();
+            _inner = null;
+        }
+
+        public bool MoveNext()
+        {
+            if (!_inner.MoveNext())
+            {
+                _current = default;
+                return false;
+            }
+
+            _current = Convert(_inner.Current);
+            return true;
+        }
+
+        public KeyValuePair<string, string[]> Current => _current;
+
+        object IEnumerator.Current => Current;
+
+        void IEnumerator.Reset() => throw new NotImplementedException();
```

Comment:
nit: change to `NotSupportedException` - `NotImplementedException` says "the library author is silly", `NotSupportedException` says "the caller is silly"

trivial example in support of this (see`void IEnumerator.Reset()`): https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABADAAgwRhQG4BYAKHIwCYcAxCCHAb3LwGY982AeASwDsALgD4cAcRiD6EAM4AKAJTNO+HMFgBDANZEAvjgPldQA==

Reply:
thanks for spotting! changed